opencog/scm/opencog/as-config_TEMPLATE.scm

```diff
@@ -0,0 +1,78 @@
+; Follow the recommendation of Guile manual for managing extensions and library paths
+; https://www.gnu.org/software/guile/manual/html_node/Modules-and-Extensions.html
+
+; We have a conditional to flick between build environment and install environment.
+
+; We call it as-config as the opencog root module is split across git repos.
+; Ideally each repo would be a separate namespace because it makes 
+(define-module (opencog as-config))
+
+; CMAKE uses this tag generate different files for in-build 
+(define-public atomspace-in-build-dir @SCM_IN_BUILD_DIR@)
+(define install-location "${CMAKE_INSTALL_PREFIX}/lib/opencog/")
+
+; If you create a new binary library for scheme, define it here
+; along with it's destination path in the build directory.
+; You also need to add the symbol to the install dir section below.
+
+(define-public opencog-ext-path-smob "@CMAKE_BINARY_DIR@/opencog/guile/")
+(define-public opencog-ext-path-exec "@CMAKE_BINARY_DIR@/opencog/atoms/execution/")
```

Comment:
This is OK-ish, but we "already" have a "more elegant" solution to this.   There is a CMakefile called `cmake/OpenCogGuile.cmake` which computes and sets these paths -- it does so generically, as a module is defined for the fist time, precisely so that no such special hand-curated lists of library paths need to be maintained. 

As I think about it, I kind-odislike this -- it risks becoming out-of-sync with the paths that `OpenCogGuile.cmake` is setting up.  If those paths get out of sync, there will be debugging confusion, since one of these two files effectively over-writes what the other one is setting up.   So it would be best to gut most of this file, and use the default paths provided by `OpenCogGuile.cmake`  instead.

(The bonus is that as-moses, agi-bio, etc. already use `OpenCogGuile.cmake` and so that way, you don't have to try to track those projects, too.)

Reply:
My reading of `cmake/OpenCogGuile.cmake` is that it does nothing with shared libraries. It only does some copying of scm files. They all just assume that their shared library will be available. There are no paths to get out of sync, because they are currently never configured! We cross our fingers and hope the library is somewhere in the RPATH chain, or in LTDL_LIBRARY_PATH (the former doesn't work with RUNPATH, the latter mixes up install and build dirs).

Having said that, it makes sense to put the configuration of shared library dependencies as an optional argument to `ADD_GUILE_MODULE`. And would simplify updating downstream projects.

I'm not a cmake expert and it'll take me a while to figure out how to run `CONFIGURE_FILE` *after* all the scm modules have been defined and stored their paths. I'll see how I go.